runtime/vm/stringhelpers.cpp

```diff
@@ -410,6 +410,81 @@ copyStringToJ9UTF8WithMemAlloc(J9VMThread *vmThread, j9object_t string, UDATA st
 	return (J9UTF8 *)result;
 }
 
+char *
+copyJ9UTF8ToUTF8WithMemAlloc(J9VMThread *vmThread, J9UTF8 *string, UDATA stringFlags, const char *prependStr, UDATA prependStrLength, char *buffer, UDATA bufferLength)
+{
+	Assert_VM_notNull(prependStr);
+	Assert_VM_notNull(string);
+
+	U_8 *result = NULL;
+	PORT_ACCESS_FROM_VMC(vmThread);
+
+	const U_8 *stringData = J9UTF8_DATA(string);
+	const UDATA stringLength = (UDATA)J9UTF8_LENGTH(string);
+	UDATA allocationSize = prependStrLength + stringLength;
```

Comment:
This is not legal C code: declarations must be at the beginning of a block. Same comment for `copyJ9UTF8WithMemAlloc()`.

Reply:
The other `copyString...` functions also are not compliant in the same way. I'll address the functions I've added here. And open a separate PR to fix the others.

Reply:
The above rule doesn't apply to C++ code. The `extern "C"` linkage specification in a C++ file indicates that the function should use C-style linkage, which affects the function name mangling but does not impose C-style variable declaration rules within the function body. Inside the function body, C++ rules apply. In C++ files, the variables should be declared as close as possible to their point of use.

Reply:
Right.

Reply:
Sorry, I thought I was looking at a `.c` file. Whether a declaration is `extern "C"` or not, is irrelevant; declarations and executable statement can be intermingled in C++ source files.